src/DocumentDB.ChangeFeedProcessor.UnitTests/LeaseManagement/DocumentServiceLeaseStoreManagerTests.cs

```diff
@@ -304,6 +305,39 @@ public async Task AcquireAsync_ThrowsLeaseLost_WhenLeaseTakenByOtherOwner()
         }
 
         [Fact]
+        public async Task AcquireAsync_DoesNotRetryConflicts_WhenAcquireReasonIsExpired()
+        {
+            var documentClient = Mock.Of<IChangeFeedDocumentClient>();
+            var cachedLease = CreateCachedLease(owner);
+            ((ILeaseAcquireReasonProvider) cachedLease).AcquireReason = LeaseAcquireReason.Expired;
+            IDocumentServiceLeaseUpdater leaseUpdater = CreateLeaseUpdater(cachedLease);
+            var leaseStoreManager = CreateLeaseStoreManager(documentClient, owner, leaseUpdater);
+
+            await leaseStoreManager.AcquireAsync(cachedLease);
+
+            Mock.Get(leaseUpdater)
+            .Verify(u => u.UpdateLeaseAsync(cachedLease, documentUri, null, It.IsAny<Func<ILease, ILease>>(), false), Times.Once);
+        }
+
+        [Theory]
+        [InlineData(LeaseAcquireReason.NotOwned)]
+        [InlineData(LeaseAcquireReason.Steal)]
+        [InlineData(LeaseAcquireReason.Unknown)]
+        public async Task AcquireAsync_RetryConflicts_WhenAcquireReasonIsNotExpired(LeaseAcquireReason leaseAcquireReason)
+        {
+            var documentClient = Mock.Of<IChangeFeedDocumentClient>();
+            var cachedLease = CreateCachedLease(owner);
+            ((ILeaseAcquireReasonProvider)cachedLease).AcquireReason = leaseAcquireReason;
+            IDocumentServiceLeaseUpdater leaseUpdater = CreateLeaseUpdater(cachedLease);
+            var leaseStoreManager = CreateLeaseStoreManager(documentClient, owner, leaseUpdater);
+
+            await leaseStoreManager.AcquireAsync(cachedLease);
+
+            Mock.Get(leaseUpdater)
+                .Verify(u => u.UpdateLeaseAsync(cachedLease, documentUri, null, It.IsAny<Func<ILease, ILease>>(), true), Times.Once);
```

Comment:
OK though I don't think Times.Once is needed here. In reality this may be not once, right?

Reply:
it is actually just once in reality. All retries happen inside UpdateLeaseAsync method